pkg/executor/guest.go

```diff
@@ -242,14 +278,56 @@ func execCommand(
 			"failed to init the executor: %v",
 			initErr,
 		)
-		return errors.New("failed to initialize command executor")
+		return errors.New("failed to initialize command executor"), nil
 	}
-	execErr := exec.Stream(remotecommand.StreamOptions{
-		Tty:    false,
-		Stdin:  ioStreams.in,
-		Stdout: ioStreams.out,
-		Stderr: ioStreams.errOut,
-	})
 
-	return execErr
+	// Setup channel for remote command execution
+	errch := make(chan error)
+
+	go func() {
+		execErr := exec.Stream(remotecommand.StreamOptions{
+			Tty:    false,
+			Stdin:  ioStreams.in,
+			Stdout: ioStreams.out,
+			Stderr: ioStreams.errOut,
+		})
+		if execErr != nil {
+			errch <- execErr
+			return
+		}
+		errch <- nil
+		return
+	}()
+
+	return nil, errch
+}
+
+// execCommandWait is a utility function for waiting for the completion of a
+// command submitted to a pod via the execCommandSync() function.
+func execCommandWait(
+	cr *kdv1.KubeDirectorCluster,
+	podName string,
+	command []string,
+	ioStreams *streams,
+	errch chan error,
+	timeInMilliSeconds time.Duration,
+) error {
+	timer := time.NewTimer(timeInMilliSeconds)
+	defer timer.Stop()
+
+	// Wait for command completion, or timeout
+	select {
+	case execErr := <-errch:
+		close(errch)
+		return execErr
+	case <-timer.C:
+		close(errch)
```

Comment:
But we'll still have a goroutine out there running the command. Hmm. I wonder if there's a way to get a handle on the underlying connection to close it.